platform/lib/routers/pages.js

```diff
@@ -56,7 +56,7 @@ function ensureFileExtension(path) {
     return path += 'index.html';
   }
 
-  if (!path.endsWith('.html')) {
+  if (!path.endsWith('.html') && !path.endsWith('.xml')) {
```

Comment:
Can we fix this more generically by only appending `.html` if there's no other file extension?

Reply:
Changed that and added unit tests for the logic.